File: python/bit strings.py
# n = int(input())
# result = 2
# t = 1 
# while 2**t < n:
#     result *= result
#     t += 1

# result *= 2**(n - 2**(t - 1))
mod = 10**9 + 7
# print(result % mod)

n = int(input())
print(2**n % mod)

File: c_c++/bit strings.cpp
# include <iostream>
using namespace std;

const int MOD = 1000000007;

long long fast_pow(long long base, long long exp, long long mod)
{
    long long result = 1;
    while (exp > 0)
    {
        if (exp % 2 == 1)
        {
            result = (result * base) % mod;
        }
        base = (base * base) % mod;
        exp /= 2;
        // cout << "exp: " << exp << endl;
    }    
    return result;
}






int main()   
{
    long long n;
    cin >> n;
    cout << fast_pow(2, n, MOD) << endl;
    return 0;
}